eks/cluster-loader/cluster-loader-add-on/addon.go

```diff
@@ -0,0 +1,96 @@
+package cluster_loader_add_on
+
+import (
+	//"encoding/json"
+	"fmt"
+	"github.com/aws/aws-k8s-tester/eksconfig"
+	k8sclient "github.com/aws/aws-k8s-tester/pkg/k8s-client"
+	gotemplate "github.com/aws/aws-k8s-tester/pkg/util"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+)
+
+type ClusterLoader struct {
+	Config    *eksconfig.Config
+	K8sClient k8sclient.EKS
+}
+
+// IsEnabled returns true if enabled
+func (c *ClusterLoader) IsEnabled() bool {
+	return c.Config.Spec.ClusterLoader != nil
+}
+
+// Apply installs the addon
+func (c *ClusterLoader) Apply() (err error) {
+	template, err := gotemplate.FromLocalDirectory(struct {
+		*eksconfig.ClusterLoaderSpec
+		ConfigMapData map[string]string
+		TestArgs      string
+	}{
+		ClusterLoaderSpec: c.Config.Spec.ClusterLoader,
+		ConfigMapData:     c.buildConfigMapData(),
+		TestArgs:          c.buildArgs(),
+	})
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	// Get Job info from goTemplate. Currently hardcoded.
+	resources := strings.Split(template.String(), "\n---\n")
+	jobYaml := resources[len(resources)-1]
+	if err := c.K8sClient.Delete(jobYaml); err != nil {
+		fmt.Errorf("while pre-deleting the job deployment: %v", err)
+	}
+	if err := c.K8sClient.Apply(template.String()); err != nil {
+		return fmt.Errorf("while applying resources, %v", err)
+	}
+	c.Config.Status.ClusterLoader = &eksconfig.ClusterLoaderStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: true,
+			Ready:     true,
+		},
+	}
+	return nil
+}
+
+// Delete removes the addon
+func (c *ClusterLoader) Delete() (err error) {
+	template, err := gotemplate.FromLocalDirectory(c.Config.Spec.ClusterLoader)
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	if err := c.K8sClient.Delete(template.String()); err != nil {
+		return fmt.Errorf("while deleting resources, %v", err)
+	}
+	c.Config.Status.ClusterLoader = &eksconfig.ClusterLoaderStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: false,
+			Ready:     false,
+		},
+	}
+	return nil
+}
+
+func (c *ClusterLoader) buildConfigMapData() (dataMap map[string]string) {
+	dataMap = make(map[string]string)
+	for _, uri := range c.Config.Spec.ClusterLoader.TestConfigUris {
+		resp, err := http.Get(uri)
+		if err != nil {
+			fmt.Errorf("while downloading uri %s: %v", uri, err)
```

Comment:
How does this work? Your function returns `dataMap map[string]string`. This shouldn't compile.

Reply:
Was just an unused error, still compiled. Fixed it to return the error now.